hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -549,6 +564,8 @@ protected void initializeOmTables(boolean addCacheMetrics)
     deletedTable = this.store.getTable(DELETED_TABLE, String.class,
         RepeatedOmKeyInfo.class);
     checkTableStatus(deletedTable, DELETED_TABLE, addCacheMetrics);
+    // Currently, deletedTable is the only table that will need the table lock
+    tableLockMap.put(DELETED_TABLE, new ReentrantReadWriteLock(true));
```

Comment:
How will we synchronize the deletes with renames ? Wouldn't you need a table level lock in the KeyTable and file table both ? can you also put some comments on what the locking hiearchy is.

Reply:
Basically any operation that read/write `deletedTable` needs to grab its table lock. It does not have a strictly enforced hierarchy like `OzoneManagerLock` does.

For the scope of this PR, only `OmSnapshotManager#createOmSnapshotCheckpoint` uses it.

Reply:
On another thought, I might be able to refactor this new table lock into another implementation of `IOzoneManagerLock`. I could look into this later.